ibis/backends/base/sqlglot/datatypes.py

```diff
@@ -447,6 +447,70 @@ class TrinoType(SqlglotType):
         }
     )
 
+    @classmethod
+    def _from_ibis_Interval(cls, dtype: dt.Interval) -> sge.DataType:
+        assert dtype.unit is not None, "interval unit cannot be None"
+        if (short := dtype.unit.short) in ("Y", "Q", "M"):
+            return sge.DataType(
+                this=typecode.INTERVAL,
+                expressions=[
+                    sge.IntervalSpan(
+                        this=sge.Var(this="YEAR"), expression=sge.Var(this="MONTH")
+                    )
+                ],
+            )
+        elif short in ("D", "h", "m", "s", "ms", "us", "ns"):
+            return sge.DataType(
+                this=typecode.INTERVAL,
+                expressions=[
+                    sge.IntervalSpan(
+                        this=sge.Var(this="DAY"), expression=sge.Var(this="SECOND")
+                    )
+                ],
+            )
+        else:
+            raise NotImplementedError(
+                f"Trino does not support {dtype.unit.name} intervals"
```

Comment:
This is implemented for the base backend, should be implemented for trino
Instead?

Reply:
Never mind, this is the trino type mapper. We should probably move these to the respective backends some point.